src/tools/segmentation/BrushTool.js

```diff
@@ -0,0 +1,135 @@
+import external from './../../externalModules.js';
+import { BaseBrushTool } from '../base';
+import store from './../../store/index.js';
+import brushUtils from '../../util/segmentation/brush/index.js';
+import EVENTS from '../../events.js';
+import { getLogger } from '../../util/logger.js';
+
+const logger = getLogger('tools:BrushTool');
+
+const { drawBrushPixels, getCircle } = brushUtils;
+
+const segmentationModule = store.modules.segmentation;
+
+/**
+ * @public
+ * @class BrushTool
+ * @memberof Tools.Brush
+ * @classdesc Tool for drawing segmentations on an image.
+ * @extends Tools.Base.BaseBrushTool
+ */
+export default class BrushTool extends BaseBrushTool {
+  constructor(props = {}) {
+    const defaultProps = {
+      name: 'Brush',
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: { alwaysEraseOnClick: false },
+    };
+
+    super(props, defaultProps);
+
+    this.touchDragCallback = this._paint.bind(this);
+  }
+
+  /**
+   * Called by the event dispatcher to render the image.
+   *
+   * @param {Object} evt - The event.
+   * @returns {void}
+   */
+  renderBrush(evt) {
+    const eventData = evt.detail;
+    const viewport = eventData.viewport;
+
+    let mousePosition;
+
+    if (this._drawing) {
+      mousePosition = this._lastImageCoords;
+    } else if (this._mouseUpRender) {
+      mousePosition = this._lastImageCoords;
+      this._mouseUpRender = false;
+    } else {
+      mousePosition = store.state.mousePositionImage;
```

Comment:
Is there an advantage to tracking this in `store` instead of in the tool's props/data?

Reply:
It was in the store already 🤷‍♂ , its set in the `mouseMove.js` event dispatcher. It seems only the freehand sculptor and brush use it. IIRC we did this in 3.0 so that tools that just wanted a canvas cursor didn't need to have their own `mouseMoveCallback` method.